src/FixedFrequencyAgent.cpp

```diff
@@ -0,0 +1,309 @@
+        for (auto &sv : m_signal_available) {
+            sv.second.m_last_signal = m_platform_io.sample(sv.second.m_batch_idx);
+        }
+    }
+
+    // Wait for the remaining cycle time to keep Controller loop cadence at 1 second
+    void FixedFrequencyAgent::wait(void)
+    {
+        geopm_time_s current_time;
+        do {
+            geopm_time(&current_time);
+        }
+        while(geopm_time_diff(&m_last_wait, &current_time) < M_WAIT_SEC);
+        geopm_time(&m_last_wait);
+    }
+
+    // Adds the wait time to the top of the report
+    std::vector<std::pair<std::string, std::string> > FixedFrequencyAgent::report_header(void) const
+    {
+        return {};
+    }
+
+    // Adds to the per-node section of the report
+    std::vector<std::pair<std::string, std::string> > FixedFrequencyAgent::report_host(void) const
+    {
+        return {};
+    }
+
+    // This Agent does not add any per-region details
+    std::map<uint64_t, std::vector<std::pair<std::string, std::string> > > FixedFrequencyAgent::report_region(void) const
+    {
+        return {};
+    }
+
+    // Adds trace columns samples and signals of interest
+    std::vector<std::string> FixedFrequencyAgent::trace_names(void) const
+    {
+        std::vector<std::string> names;
+        // Build name in the format: "NVML::FREQUENCY-board-0"
+        for (auto &sv : m_signal_available) {
+            names.push_back(sv.first + "-" + m_platform_topo.domain_type_to_name(GEOPM_DOMAIN_BOARD) + "-0");
+        }
+        return names;
+    }
+
+    // Updates the trace with values for samples and signals from this Agent
+    void FixedFrequencyAgent::trace_values(std::vector<double> &values)
+    {
+        int values_idx = 0;
+        // Default assumption is that every signal added should be in the trace
+        for (auto &sv : m_signal_available) {
+            values[values_idx] = sv.second.m_last_signal;
+            ++values_idx;
+        }
+    }
+
+    std::vector<std::function<std::string(double)> > FixedFrequencyAgent::trace_formats(void) const
+    {
+        std::vector<std::function<std::string(double)>> trace_formats;
+        for (auto &sv : m_signal_available) {
+            trace_formats.push_back(m_platform_io.format_function(sv.first));
+        }
+        return trace_formats;
+    }
+
+    // Name used for registration with the Agent factory
+    std::string FixedFrequencyAgent::plugin_name(void)
+    {
+        return "fixed_frequency";
+    }
+
+    // Used by the factory to create objects of this type
+    std::unique_ptr<Agent> FixedFrequencyAgent::make_plugin(void)
+    {
+        return geopm::make_unique<FixedFrequencyAgent>();
+    }
+
+    // Describes expected policies to be provided by the resource manager or user
+    std::vector<std::string> FixedFrequencyAgent::policy_names(void)
+    {
+        return {"ACCELERATOR_FREQUENCY",
+                "CORE_FREQUENCY",
+                "UNCORE_FREQUENCY",
```

Comment:
Do we want to provide the user the ability to input a min & max uncore frequency?

Reply:
Updated to have UNCORE_MIN_FREQUENCY and UNCORE_MAX_FREQUENCY